Runtime/Core/ProBuilderMeshFunction.cs

```diff
@@ -368,6 +370,10 @@ public void ToMesh(MeshTopology preferredTopology = MeshTopology.Triangles)
 
             mesh.subMeshCount = submeshes.Length;
 
+            m_MeshMaterials.Clear();
+            renderer.GetSharedMaterials(m_MeshMaterials);
```

Comment:
instead of adding a new static list to `ProBuilderMesh`, you could re-use `MaterialUtility.s_MaterialArray` as a temporary buffer.

Reply:
Gotcha! Modified in [3772d72](https://github.com/Unity-Technologies/com.unity.probuilder/pull/460/commits/3772d72255c8887841c75887d34a1c1d4e104775)